core/node/node_sync/src/external_io.rs

```diff
@@ -155,6 +155,14 @@ impl StateKeeperIO for ExternalIO {
                 )
             })?;
         let Some(mut pending_l2_block_header) = pending_l2_block_header else {
+            tracing::info!(
+                l1_batch_number = %cursor.l1_batch,
+                "Empty unsealed batch found; deleting it as we need at least one L2 block to initialize"
```

Comment:
Dumb question: Does an unsealed batch necessarily exist at this point? AFAICT, `delete_unsealed_l1_batch` is a no-op if it doesn't (so it would work correctly), but the log message may be misleading.

Reply:
Good catch, clarified